File: solutions/0009-palindrome-number/solution.cpp
class Solution {
public:
    bool isPalindrome(int x) {
       int X = x;
       long r = 0;
       while(x > 0){
         r +=  x%10;
         r *= 10;
         x /= 10;
        }
       r /= 10;
       if(X == r){
        return true;
       }else{
        return false;
       }
    }
};

File: solutions/0009-palindrome-number/solution.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0){
            return false;
        }
        int x2 = x;
        int rev = 0;
        while(x2 != 0){
            rev = rev*10 + x2%10;
            x2 /= 10;
        }
        if(rev == x){
            return true;
        }else{
            return false;
        }
    }
}
